src/Nirum/Targets/Python.hs

```diff
@@ -1033,35 +1037,48 @@ compileTypeDeclaration src
     typeRepr <- typeReprCompiler
     ret <- returnCompiler
     arg <- parameterCompiler
-    return [qq|
-class $className({T.intercalate "," $ compileExtendClasses annotations}):
-{compileDocstring "    " d}
+    return $ renderCompileText $ [compileText|
+class #{className}(#{T.intercalate "," $ compileExtendClasses annotations}):
+#{compileDocstring "    " d}
 
     __nirum_type__ = 'union'
-    __nirum_union_behind_name__ = '{I.toSnakeCaseText $ N.behindName typename'}'
-    __nirum_field_names__ = name_dict_type([$nameMaps])
+    __nirum_union_behind_name__ = '#{toBehindSnakeCaseText typename'}'
+    __nirum_field_names__ = name_dict_type([
+%{ forall (Tag (Name f b) _ _) <- tags }
+        ('#{toAttributeName f}', '#{I.toSnakeCaseText b}'),
+%{ endforall }
+    ])
 
     class Tag(enum.Enum):
-        $enumMembers
+%{ forall (Tag tn _ _) <- tags }
+        #{toEnumMemberName tn} = '#{toBehindSnakeCaseText tn}'
+%{ endforall }
 
     def __init__(self, *args, **kwargs):
         raise NotImplementedError(
-            "\{0\} cannot be instantiated "
+            "{0} cannot be instantiated "
             "since it is an abstract class.  Instantiate a concrete subtype "
-            "of it instead.".format({typeRepr "type(self)"})
+            "of it instead.".format(#{typeRepr "type(self)"})
         )
 
     def __nirum_serialize__(self):
         raise NotImplementedError(
-            "\{0\} cannot be instantiated "
+            "{0} cannot be instantiated "
             "since it is an abstract class.  Instantiate a concrete subtype "
-            "of it instead.".format({typeRepr "type(self)"})
+            "of it instead.".format(#{typeRepr "type(self)"})
         )
 
     @classmethod
     def __nirum_deserialize__(
-        {arg "cls" "type"}, value
-    ){ ret className }:
+        #{arg "cls" "type"}, value
+    )#{ ret className }:
```

Comment:
Since we now have proper conditionals through Heterocephalus, things like `arg "cls" "type"` or `ret className` can be avoided.  See also `compileTypeDeclaration` function for `EnumType`:

```
    @classmethod
%{ case pyVer }
%{ of Python2 }
    def __nirum_deserialize__(cls, value):
%{ of Python3 }
    def __nirum_deserialize__(cls: type, value: str) -> '#{className}':
%{ endcase }
        return cls(value.replace('-', '_'))  # FIXME: validate input
```